Sources/AblyChat/ChatAPI.swift

```diff
@@ -0,0 +1,232 @@
+import Ably
+
+final class ChatAPI: Sendable {
+    private let rest: ARTRest
+    private let realtime: RealtimeClient
+    private let apiProtocolVersion: Int = 3
+    
+    init(rest: ARTRest, realtime: RealtimeClient) {
+        self.rest = rest
+        self.realtime = realtime
+    }
+
+    func getMessages(roomId: String, params: QueryOptions) async throws -> any PaginatedResult<Message> {
+        let endpoint = "/chat/v1/rooms/\(roomId)/messages"
+        let response: any PaginatedResult<Message> = try await makeAuthorizedPaginatedRequest(endpoint, params: params.toDictionary())
+        return response
+    }
+    
+    struct SendMessageResponse: Codable {
+        let timeserial: String
+        let createdAt: Int64
+    }
+
+    func sendMessage(roomId: String, params: SendMessageParams) async throws -> Message {
+        let endpoint = "/chat/v1/rooms/\(roomId)/messages"
+        var body: [String: Any] = ["text": params.text]
+        
+        if let metadata = params.metadata {
+            body["metadata"] = metadata
+        }
+        
+        if let headers = params.headers {
+            body["headers"] = headers
+        }
+        
+        let response: SendMessageResponse = try await makeAuthorizedRequest(endpoint, method: "POST", body: body)
+        let timeIntervalInSeconds = TimeInterval(integerLiteral: response.createdAt) / 1000
+        
+        let message = Message(
+            timeserial: response.timeserial,
+            clientID: realtime.clientId ?? "",
+            roomID: roomId,
+            text: params.text,
+            createdAt: Date(timeIntervalSince1970: timeIntervalInSeconds),
+            metadata: params.metadata ?? [:],
+            headers: params.headers ?? [:]
+        )
+        return message
+    }
+
+    func getOccupancy(roomId: String) async throws -> OccupancyEvent {
+        let endpoint = "/chat/v1/rooms/\(roomId)/occupancy"
+        return try await makeAuthorizedRequest(endpoint, method: "GET")
+    }
+    
+    private func makeAuthorizedRequest<RES: Codable>(_ url: String, method: String, body: [String: Any]? = nil) async throws -> RES {
+            return try await withCheckedThrowingContinuation { continuation in
```

Comment:
It would be good if this `ChatAPI` class could focus on the details of the Chat API, and not the bridging with ably-cocoa’s callback-based interface. I’d suggest taking an approach similar to the `RoomLifecycleContributorChannel` that I added in #54 — that is, create a protocol that provides an `async` version of `request` (and then create a conforming wrapper for an ably-cocoa realtime client). And then you could also make this version of `request` return a paginated result that conforms to Chat’s `async`-based `PaginatedResult` protocol.

Reply:
let's save that as a next step, I'll look into when picking up the reactions stuff. would rather get more functionality done atm (this is also a private func so doesn't impact usability of the SDK itself)